### DISCUSSION TITLE
How can we tell if an Octopus Savings Session has been joined?

### DISCUSSION THREAD

**Discussion by jameslavery** [springfall2008/batpred#627] [Q&A] [answered]
I know the system will auto-join sessions.

Is it possible to tell from the UI/output that a session has been joined? Or is it just a case of keeping an eye on the Plan?

**Comment by marksie1988** [comment]
I got a notification on my phone when it joined

**Reply by jameslavery** [reply]
Was that from Octopus, or from HA?

**Reply by marksie1988** [reply]
That was from HA it told me that batpred had signed me up to a saving session

**Comment by springfall2008** [comment, marked as answer]
There’s a notification, you can see the rates on chart change, it’s mentioned in the log file or you can use the octopus app to confirm

**Reply by jameslavery** [reply]
Rate changes etc. are a good indicator - yes. Also I then did check on the Octopus app and it said I'd enrolled. That's good enough for me!

I was just wondering if BatPred indicated explicitly that a Savings Session had been signed up to.

When you say 'notification', is that a HA notification - if I've set HA to give notifications? I'm new to HA, and getting to grips with its features!

**Reply by springfall2008** [reply]
Yes a HA notification is generated, this goes to your mobile devices that are linked to the App.

**Reply by gcoan** [reply]
@jameslavery if you have the Home Assistant Companion app installed on your mobile, that's how you get notifications from Predbat when saving sessions are automatically joined

**Reply by pacemaker82** [reply]
@jameslavery I am checking this via the binary_sensor.octopus_energy_a_c{x}_octoplus_saving_sessions sensor from the Octopus integration (not PredBat). If you check the attributes of the sensor you can see the "next" charge start/end time attributes are set when one is coming. This sensor will return true when the session is active. 

I too would like an explicit boolean true/false if one is signed up to though. Would be a good addition to PredBat.

**Reply by gcoan** [reply]
You can always create a custom template sensor in HA to read the attribute value using state_attr()

This would be a better solution than adding an extra sensor that's populated infrequently (and repeats other data already in HA).

Have a look on youtube for videos on custom template sensors - smarthomejunkie ones are quite good
